gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/components/jdbc/VertxJdbcClientImpl.java

```diff
@@ -25,24 +28,27 @@
 import io.vertx.ext.jdbc.JDBCClient;
 import io.vertx.ext.jdbc.spi.impl.HikariCPDataSourceProvider;
 
-import java.util.Map.Entry;
-
 import javax.sql.DataSource;
+import java.util.Map.Entry;
 
 /**
  * @author Marc Savy {@literal <msavy@redhat.com>}
  */
 public class VertxJdbcClientImpl implements IJdbcClient {
-    //TODO cache jdbcClients && json configs? LinkedHashMap + removeEldestEntry?
-    //private static Map<>
-    private JDBCClient jdbcClient;
+    private static final Cache<String, JDBCClient> cache = Caffeine.newBuilder()
+            .maximumSize(20)
```

Comment:
Is 20 just a magic number?

Reply:
Yes, I thought it would maybe make scene to make this configurable but not sure if there is really a need for it.
Currently it would cache 20 clients (20 different db configs, I am not sure if we really have the need for more.)
If you would use the same db table for different basic-auth-policy configs it would be always the same client.